events/teamJoin.js

```diff
@@ -0,0 +1,17 @@
+function teamJoin(context, payload) {
+  const playerState = context.getPlayerState();
+  const playerId = playerState.id;
+  const game = context.getGameState();
+  const teamName = game.teams[payload];
+
+  playerState.team = payload;
```

Comment:
Now your working with the player state of the server state, that is not the state of a game. So all players and what team they belong to are stored within a game (this might be confusing).  Instead you could do something like this:
``` js
const playerId = context.id(); // ourselves
const game = context.getGameState(); // the game
// This is what is different from your code
const player = game.players[playerId]; // This is how we get a user from within a game
// Then we can change what team they belong to.
// Also notice we get the key `team` from the payload, since all payloads are always an object.
player.team = payload.team;
// We would then update the game with this player
game.players[playerId] = player;

// Another way of easily doing it could be
game.players[playerId].team = payload.team;
```

Reply:
This made things a lot clearer! Thank you!!